core/vm/privacy/ringct.go

```diff
@@ -91,7 +91,20 @@ func SerializeCompressed(p *ecdsa.PublicKey) []byte {
 }
 
 func DeserializeCompressed(curve elliptic.Curve, b []byte) *ecdsa.PublicKey {
+	if len(b) < 33 {
+		// Not enough data to represent a compressed public key
+		return nil
+	}
+
 	x := new(big.Int).SetBytes(b[1:33])
+	if x.Sign() == 0 {
```

Comment:
would something like this be too complex for a check? taken from ringct_test
add this point to an arbitrary point on curve
if the result doesn't change, then this point is (0,0)
```
func TestCurveAddZero(t *testing.T) {
	// curve := crypto.S256()
	curve := crypto.S256().(*secp256k1.BitCurve)

	x1, y1 := curve.ScalarBaseMult(new(big.Int).SetUint64(uint64(1)).Bytes())
	fmt.Printf("Point(%x, %x)\n", x1, y1)

	x0 := new(big.Int).SetUint64(uint64(0))
	y0 := new(big.Int).SetUint64(uint64(0)) // infinity
	fmt.Printf("Is point (%d,%d) on the curve: %t \n", x0, y0, curve.IsOnCurve(x0, y0))

	x2, y2 := curve.Add(x1, y1, x0, y0)
	fmt.Printf("Output is Point(%x, %x)\n", x2, y2)

	if (x1.Cmp(x2) == 0) && (y1.Cmp(y2) == 0) {
		// fmt.Printf("Correct, Point on curve is the same after Zero addition\n")
	} else {
		t.Error("Incorrect, Point on curve changed after Zero addition\n")
	}
}
```

Reply:
hi @wanwiset25 , could you elaborate a bit more? i'm not sure i'm following, what's too complex for a check?

Reply:
The operation of checking addition with arbitrary point. Just thinking maybe it has big overhead.